### PR TITLE
[RELEASE] Release opentelemetry-cpp 1.17.0

### LINKED ISSUE #3069: [RELEASE] Prepare next release
No planned date.

Tentative content:

* #3044
* #3041

### PR DESCRIPTION
**Pull request by marcalff**
Fixes #3069

## Changes

Please provide a brief description of the changes here.

* Release opentelemetry-cpp 1.17.0

For significant contributions please make sure you have completed the following items:

* [X] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [X] Changes in public API reviewed

### CODE REVIEW

**Comment by marcalff** [comment]
@open-telemetry/cpp-maintainers,

Please review @owent PR as well, so it can do in the release:

* #3041

**Comment by marcalff** [comment]
On hold one week, to review GRPC changes.

**Review by lalitb — APPROVED** [review]

**Pull request merged**
